### PR TITLE
Updated build script and added assembly details to the C++CLI projects

### LINKED ISSUE #562: No versions on CefSharp.Core and CefSharp.BrowserSubprocess.Core projects
We rely on DLL versions to determine if we have to deploy a file or not. `CefSharp.Core` and `CefSharp.BrowserSubprocess.Core` don't have version. For now, we fixed it in https://github.com/Octopus-ITSM/CefSharp/commit/0c5520e20032c5d5e7ad8e73904fad9d6d4b1ff4 but I don't want to send you a PR for this change because I'm not sure this is how you want to handle bumping the versions.

### LINKED ISSUE #692: Build Script Update App.Manifest
Have the build script automatically update the version in `app.manifest` files.

https://github.com/cefsharp/CefSharp/blob/master/CefSharp.BrowserSubprocess/app.manifest#L11
https://github.com/cefsharp/CefSharp/blob/master/CefSharp.WinForms.Example/app.manifest#L11

### PR DESCRIPTION
**Pull request by merceyz**
Resolves #692 and resolves #562

- Made the build script upate the version in app.manifest files

- Build script now updates AssemblyFileVersion as well 

- Added resource files to the C++ CLI projects to be able to set the assembly details

- Made the build script update the version in said resource files

### CODE REVIEW

**Comment by amaitland** [comment]
Nice one 👍 

Hopefully look at this next week.

**Comment by merceyz** [comment]
For some reason GitHub moved my commit from another branch into master when said branch was deleted.
Just ignore those last two commits, once this PR is complete they'll show up in another PR

File: CefSharp.Wpf/ChromiumWebBrowser.cs [hunk 1]
```diff
@@ -1593,10 +1593,16 @@ public override void OnApplyTemplate()
         {
             base.OnApplyTemplate();
 
-            // Create main window
-            Content = image = CreateImage();
+            if (image == null)
```
Comment (amaitland) [inline CefSharp.Wpf/ChromiumWebBrowser.cs]: Seems unrelated to the PR

Comment (merceyz) [inline CefSharp.Wpf/ChromiumWebBrowser.cs]: It is, look at https://github.com/cefsharp/CefSharp/pull/1947#issuecomment-280770217

**Comment by amaitland** [comment]
Better to create a branch for each set of changes your working on

**Comment by merceyz** [comment]
I will remember that for the next time

**Comment by amaitland** [comment]
Merged, thanks 👍
